### DISCUSSION TITLE
this learning section feels messy

### DISCUSSION THREAD

**Discussion by toastdoteth2** [Cyfrin/foundry-full-course-cu#2332] [General]
First wanna say that I'm really thankful for this course. I'm learning a lot!

Right now I'm up to Foundry Fundamentals - 20. Integrations Tests but its a lot. Patrick is going through doing a lot of copying and pasting 'this from here and into here', constantly renaming functions etc etc.. its hard to follow along. 

At this point I'm relying on the repeating of copying code snippets to have an impact on me learning whats happening than actually understanding what he's actually doing but if the rest of the course continues like this, then I dont think I'll be retaining a whole lot. I'm hopeful that its just this section!

**Comment by EngrPips** [comment]
Hello @toastdoteth2, The other section of the course has limited copying and pasting or moving files from one place to another. Just try your best to hold it down through that section. Gradually, copying and moving files from one place to another will make more sense to you. Make sure to redo any section that doesn't seem clear to you, and also don't hesitate to ask questions on here to clarify stuff
